flink-end-to-end-tests/test-scripts/kafka-common.sh

```diff
@@ -37,7 +37,7 @@ MAX_RETRY_SECONDS=120
 function setup_kafka_dist {
   # download Kafka
   mkdir -p $TEST_DATA_DIR
-  KAFKA_URL="https://archive.apache.org/dist/kafka/$KAFKA_VERSION/kafka_2.12-$KAFKA_VERSION.tgz"
+  KAFKA_URL="https://downloads.apache.org/kafka/$KAFKA_VERSION/kafka_2.12-$KAFKA_VERSION.tgz"
```

Comment:
Does this mean that when the Kafka version we are using is no longer recommended, we must forcibly upgrade it. I'm not sure if we're currently syncing updates.

Reply:
I believe we are keeping up with Kafka upgrades anyway, as its a fundamental component. If we want to stick to older version in the future we can always switch back to the archive (hoping the slow-down is resolved, or bumping the timeout)

Reply:
> I believe we are keeping up with Kafka upgrades anyway, as its a fundamental component.

An upgrade must be deliberate on the Flink side; for example, if Kafka pushes a faulty release, we might want to skip it, but the older one would no longer be available.

Also, we need to have some headroom for the upgrade without being stressed about all builds suddenly failing.